evm/src/keccak_sponge/keccak_sponge_stark.rs

```diff
@@ -49,15 +49,34 @@ pub(crate) fn ctl_looked_data<F: Field>() -> Vec<Column<F>> {
 
 pub(crate) fn ctl_looking_keccak<F: Field>() -> Vec<Column<F>> {
     let cols = KECCAK_SPONGE_COL_MAP;
-    Column::singles(
+    let mut res: Vec<_> = Column::singles(
         [
             cols.xored_rate_u32s.as_slice(),
             &cols.original_capacity_u32s,
-            &cols.updated_state_u32s,
         ]
         .concat(),
     )
-    .collect()
+    .collect();
+
+    // We recover the 32-bit digest limbs from their corresponding bytes,
+    // and then append them to the rest of the updated state limbs.
+    let digest_u32s = cols
+        .updated_digest_state_bytes
+        .chunks_exact(4)
+        .map(|c| {
+            Column::linear_combination(
+                c.iter()
+                    .enumerate()
+                    .map(|(i, &b)| (b, F::from_canonical_usize(1 << (8 * i)))),
+            )
+        })
+        .collect::<Vec<_>>();
+
+    res.extend(digest_u32s);
```

Comment:
For both of these cases, I think we can call `extend` without `collect`ing first?